### PR TITLE
Dc status

### PR DESCRIPTION
**Pull request by vinay-ebi**
Retrieve datachecks job status for given job id ({'total': 0, 'inprogress': 0, 'completed': 0, 'failed': 0 }) which used by handover app to display dc job progress.

### CODE REVIEW

File: src/ensembl/production/core/models/hive.py [hunk 1]
```diff
@@ -327,13 +330,43 @@ def get_result_for_job(self, job, progress=False):
             else:
                 result['status'] = self.get_job_tree_status(job)
                 if progress:
-                    result['progress'] = self.get_jobs_progress(job)
+                    result['progress'] = self.get_all_jobs_progress(job.job_id, analysis_id=analysis_id)
+                    #result['progress'] = self.get_jobs_progress(job)
```
Comment (marcoooo) [inline src/ensembl/production/core/models/hive.py]: Could you remove as much as possible commented code? it's often misleading for the one who will come after you.

Comment (vinay-ebi) [reply src/ensembl/production/core/models/hive.py]: sure i will remove the comment code

File: src/ensembl/production/core/models/hive.py [hunk 2]
```diff
@@ -327,13 +330,42 @@ def get_result_for_job(self, job, progress=False):
...
                 if progress:
-                    result['progress'] = self.get_jobs_progress(job)
+                    result['progress'] = self.get_all_jobs_progress(job.job_id, analysis_id=analysis_id)
         except ValueError as e:
             raise ValueError('Cannot retrieve results for job: {}'.format(job.job_id)) from e
         except SQLAlchemyError as e:
             raise ValueError('DB error for job: {}'.format(job.job_id)) from e
         return result
 
+    def get_all_jobs_progress(self, job_id, analysis_id=None):
```
Comment (james-monkeyshines) [inline src/ensembl/production/core/models/hive.py:340]: I don't quite understand what the updated code is doing in this module. An `analysis_id` parameter has been introduced, but the updated code in https://github.com/Ensembl/ensembl-prodinf-datachecks/pull/9 doesn't pass a value for it, so does not seem like it's being used.
In any case, an analysis_id is less specific than a job_id; so filtering on both is the same as filtering on just the job_id.

I think the aim is to indicate the status of individual datachecks. Which is difficult to determine with any degree of certainty. It's possible for a single job to run multiple datachecks (the 'RunDataChecks' analysis), or they could be running in parallel (the 'DataCheckFan' analysis) - but in the latter case, the datachecks can't really be considered to be completed until the 'DataCheckFunnel' analysis has merged them back together, and written the output and history to disk. So there is no single 'analysis' that can be used to track progress, and anyway the processes are not really defined at the level of individual datachecks.

Without a lot more development, I think the only workable apporach is to fetch the status of all children of a given job_id, and present the percentages for done/failed/running. It won't tell you exactly how many datachecks are left to run, but will give an approximation (with the caveat that a few final jobs may take several hours, so the progress could quickly get to 90%, say, and then stay at that value for a while).

Comment (vinay-ebi) [reply src/ensembl/production/core/models/hive.py:340]: Yes,  analsys_id is currently not used but as this is generalised hive model i have provided with this option for future usage and your right filtering with job_id is same as less specific analysis_id. My intention is to provide a option to get all jobs for given analysis_id.  I will change the query accordingly to get all jobs for given analysis_id if job_id is not given.

I am querying all  Datacheck child jobs using parent job id from job table using param_id_stack column irrespective of analysis and provide with absolute count for   done/failed/running status with out progress bar. Below image will give more clarity on what i intended to do. 
image

Comment (james-monkeyshines) [reply src/ensembl/production/core/models/hive.py:340]: OK, thanks for that, I understand much better now.

**Review by james-monkeyshines — APPROVED** [review]

**Review by marcoooo — APPROVED** [review]

**Pull request merged**